scripts/run_performance_tests.py

```diff
@@ -0,0 +1,241 @@
+        ),
+    )
+    parser.add_argument(
+        '-m',
+        '--mode',
+        choices=[
+            value.name
+            for value in TestCase
+        ],
+        required=True,
+        help='Specifies the type of content to generate.',
+    )
+    parser.add_argument(
+        '-L',
+        '--length',
+        type=assert_positive(int),
+        help='Length of test case content.',
+    )
+
+    return parser.parse_args()
+
+
+def assert_positive(type):
+    def wrapped(string):
+        value = type(string)
+        if value <= 0:
+            raise argparse.ArgumentTypeError(
+                '{} must be a positive {}.'.format(
+                    string,
+                    type.__name__,
+                ),
+            )
+
+        return value
+
+    return wrapped
+
+
+def assert_valid_file(string):
+    if not os.path.isfile(string):
+        raise argparse.ArgumentTypeError(
+            '{} must be a valid file.'.format(string),
+        )
+
+    with open(string) as f:
+        return json.load(f)
+
+
+def generate_test_content(mode, **kwargs):
+    """
+    :type mode: TestCase
+    :type length: int
+    :type timeout: float
+    """
+    if not kwargs['length']:
+        del kwargs['length']
+
+    print('Generating content...', file=sys.stderr)
+    if mode == TestCase.LONG_FILE:
+        return generate_long_file(**kwargs)
+    elif mode == TestCase.LONG_LINES:
+        return generate_long_lines(**kwargs)
+
+
+def scan_content(content, timeout, baseline=None):
+    """
+    :type content: str
+    :type timeout: float
+    :type baseline: dict|None
+    """
+    args = [
+        'python',
+        os.path.join(
+            get_root_directory(),
+            'scripts/benchmark.py',
+        ),
+        '--harakiri', str(timeout),
+    ]
+
+    with tempfile.NamedTemporaryFile('w') as f:
+        f.write(content)
+
+        print('Running checks...', file=sys.stderr)
+        if not baseline:
+            args.append(f.name)
+            return subprocess.check_output(
+                args,
+                stderr=subprocess.DEVNULL,
+            ).decode('utf-8')
+
+        with tempfile.NamedTemporaryFile('w') as b:
+            b.write(
+                json.dumps({
+                    'filenames': [f.name],
+                    'timings': baseline['timings'],
+                }),
+            )
+            b.seek(0)
+
+            args.append('--baseline')
+            args.append(b.name)
+
+            return subprocess.check_output(
+                args,
+                stderr=subprocess.DEVNULL,
+            ).decode('utf-8')
+
+
+def generate_long_file(length=250000, **kwargs):
+    return generate_content(
+        separator='\n',
+        length=length,
+    )
+
+
+def generate_long_lines(length=250000, **kwargs):
+    return generate_content(
+        separator=' ',
+        length=length,
+    )
+
+
+def generate_content(separator, length):
```

Comment:
@killuazhu , with regards to https://github.com/Yelp/detect-secrets/issues/136, can you see what changes this code needs to undergo in order to reproduce your performance hit? Even with my sample of 250K lines, it only took ~16s.

Reply:
@domanchi  The performance hit was resolved with the short term fix in https://github.com/Yelp/detect-secrets/pull/144 and https://github.com/Yelp/detect-secrets/pull/147. If we want to reproduce, we will need to revert those 2 PRs.

Reply:
Thanks for the convenient links!

After reverting those changes locally, I get these results:

```
$ scripts/run_performance_tests.py -m LONG_FILE --baseline baseline.json
Generating content...
Running checks...
------------------------------------------------------------
plugin                          time                change
------------------------------------------------------------
All Plugins               Timeout exceeded!        ▲  13.9
AWSKeyDetector                 1.38776s            ▲  0.02
ArtifactoryDetector            1.87043s            ▲  0.03
Base64HighEntropyString   Timeout exceeded!       ▲  27.11
BasicAuthDetector              1.37341s            ▼ -0.02
HexHighEntropyString      Timeout exceeded!       ▲  27.13
KeywordDetector                2.47949s            ▼ -0.02
PrivateKeyDetector             1.85911s               -
SlackDetector                  1.5571s             ▼ -0.06
StripeDetector                 1.40493s               -
------------------------------------------------------------
```

Good to know our changes were effective, and that this script would help us identify any further issues in the future!